src/sources/datadog/agent/mod.rs

```diff
@@ -75,19 +55,15 @@ struct DatadogAgentConfig {
     #[serde(default, deserialize_with = "bool_or_struct")]
     acknowledgements: AcknowledgementsConfig,
     #[serde(default = "crate::serde::default_false")]
+    disable_logs: bool,
+    #[serde(default = "crate::serde::default_false")]
+    disable_metrics: bool,
+    #[serde(default = "crate::serde::default_false")]
+    disable_traces: bool,
+    #[serde(default = "crate::serde::default_false")]
     multiple_outputs: bool,
 }
 
-inventory::submit! {
-    SourceDescription::new::<DatadogAgentConfig>("datadog_agent")
-}
```

Comment:
Just a note that this got dropped causing #12469 

I'm surprised it didn't seem to cause more issues though 🤔